cmd/tsidp/tsidp.go

```diff
@@ -183,16 +206,29 @@ func main() {
 		srv.serverURL = fmt.Sprintf("https://%s", strings.TrimSuffix(st.Self.DNSName, "."))
 	}
 
-	// Load funnel clients from disk if they exist, regardless of whether funnel is enabled
-	// This ensures OIDC clients persist across restarts
-	f, err := os.Open(funnelClientsFile)
+	// Load OAuth/funnel clients from disk if they exist
```

Comment:
// If allowInsecureRegistration is enabled, the old oidc-funnel-clients.json path is used. If allowInsecureRegistration is disabled, attempt to migrate the old path to oidc-clients.json and use this new path.